Source/WebCore/platform/graphics/skia/SkiaThreadedPaintingPool.cpp

```diff
@@ -39,17 +43,28 @@ namespace WebCore {
 
 WTF_MAKE_TZONE_ALLOCATED_IMPL(SkiaThreadedPaintingPool);
 
-SkiaThreadedPaintingPool::SkiaThreadedPaintingPool(unsigned numberOfThreads)
-    : m_workerPool(WorkerPool::create("SkiaPaintingThread"_s, numberOfThreads))
+// Note:
+// If WEBKIT_SKIA_ENABLE_CPU_RENDERING is unset, we will allocate a GPU-only worker pool with WEBKIT_SKIA_GPU_PAINTING_THREADS threads (default: 1).
+// If WEBKIT_SKIA_ENABLE_CPU_RENDERING is unset, and WEBKIT_SKIA_GPU_PAINTING_THREADS is set to 0, we will use GPU rendering on main thread.
+//
+// If WEBKIT_SKIA_ENABLE_CPU_RENDERING=1 is set, we will allocate a CPU-only worker pool with WEBKIT_SKIA_CPU_PAINTING_THREADS threads (default: nCores/2).
+// if WEBKIT_SKIA_ENABLE_CPU_RENDERING=1 is set, and WEBKIT_SKIA_CPU_PAINTING_THREADS is set to 0, we will use CPU rendering on main thread.
+
+SkiaThreadedPaintingPool::SkiaThreadedPaintingPool(unsigned numberOfCPUThreads, unsigned numberOfGPUThreads)
 {
+    // By default, GPU rendering, if activated, takes precedence over CPU rendering.
+    if (ProcessCapabilities::canUseAcceleratedBuffers()) {
+        m_texturePool = makeUnique<BitmapTexturePool>();
+
+        if (numberOfGPUThreads > 0)
```

Comment:
```suggestion
        if (numberOfGPUThreads)
```

Reply:
Well it can be null, in that case we want to avoid creating the worker pool.

Reply:
Heh, forget about it. It's unsigned, so this is equivalent.

Reply:
Fixed.